lib/pedicel/ec.rb

```diff
@@ -6,10 +6,15 @@ def ephemeral_public_key
       Base64.decode64(@token['header']['ephemeralPublicKey'])
     end
 
+    # Decrypt self.encrypted_data() finding the key from either
+    # symmetric_key, og
+    #
     def decrypt(symmetric_key: nil, merchant_id: nil, certificate: nil, private_key: nil,
-                ca_certificate_pem: @config[:apple_root_ca_g3_cert_pem], now: Time.now)
-      raise ArgumentError 'invalid argument combination' unless \
-        !symmetric_key.nil? ^ ((!merchant_id.nil? ^ !certificate.nil?) && !private_key.nil?)
+                ca_certificate_pem: @config[:trusted_ca_pem], now: Time.now)
+      raise Pedicel::ArgumentError, 'invalid argument combination' unless \
```

Comment:
I think it's perfectly fine to raise an `ArgumentError`; one wouldn't want to catch such an error (I think) when rescuing `Pedicel::Error`s?

Reply:
I agree! I'd missed the builtin ArgumentError :/